### PR TITLE
V2 TO V3 fragment changes for extension and server directory under test/.. 

### LINKED ISSUE #10843: Convert v2 API test fragments to v3
While all programatically constructed test configs will be v3 (via API boosting rewriting), there are still v2 JSON/YAML fragments in test code. Many of these will "just work" as v3, but some may have v2 features that will have them treated as v2. This becomes confusing when they are extended by developers who want to model v3 features.

These need to be audited and updated as appropriate. Also, the names of methods such as `parseAccessLogFromV2Yaml` should be updated to reflect v3 when operating on fragments that should be v3.

This is somewhat similar to the work done by @dmitri-d for examples in https://github.com/envoyproxy/envoy/pull/10644 as part of https://github.com/envoyproxy/envoy/issues/9735.

CC @alyssawilk

### PR DESCRIPTION
**Pull request by ankatare**
Signed-off-by: Abhay Narayan Katare <abhay.katare@india.nec.com>

Commit Message: v2 to v3 fragment changes for extension and server directory test cases.
Additional Description: 
Risk Level: Low
Testing: unit and format
Docs Changes:NA
Release Notes: NA
[Optional Runtime guard:]
Optional Fixes #10843 
[Optional Deprecated:]

### CODE REVIEW

**Comment by ankatare** [comment]
@junr03 seems some env issues in this PR. 
https://dev.azure.com/cncf/envoy/_build/results?buildId=44878&view=results

please suggest if it is good to go with code review.

**Comment by junr03** [comment]
Try merging master again?

**Comment by ankatare** [comment]
@junr03 ok.

**Comment by ankatare** [comment]
@junr03 merged from master. again seems some different env issues. please suggest if it is ok to review code.

**Comment by ankatare** [comment]
@junr03 sure, please suggest how to fix it , any pointers would be helpful.

**Comment by ankatare** [comment]
@junr03 i tried below command to fix issue 
clang-tidy -fix-errors -checks='readability-identifier-naming' test/extensions/filters/network/thrift_proxy/filters/ratelimit/ratelimit_test.cc.
That fixes clang-tidy error. but while unit testing it with bazel test it is failing locally.
would you please suggest what to do as in this case as i can't push test file now.

**Comment by junr03** [comment]
@ankatare what is the failure trace?

File: test/extensions/filters/network/thrift_proxy/filters/ratelimit/ratelimit_test.cc [hunk 1]
```diff
@@ -93,7 +93,7 @@ class ThriftRateLimitFilterTest : public testing::Test {
   NiceMock<LocalInfo::MockLocalInfo> local_info_;
 };
 
-TEST_F(ThriftRateLimitFilterTest, NoRoute) {
+testF(ThriftRateLimitFilterTest, NoRoute) {
```
Comment (junr03) [inline test/extensions/filters/network/thrift_proxy/filters/ratelimit/ratelimit_test.cc:96]: what is `testF`?

Comment (ankatare) [reply test/extensions/filters/network/thrift_proxy/filters/ratelimit/ratelimit_test.cc:96]: @junr03 i ran clang-tidy -fix-errors -checks='readability-identifier-naming' < file> to fix errors and it renamed function. let me know if i need to revert it back.

Comment (ankatare) [reply test/extensions/filters/network/thrift_proxy/filters/ratelimit/ratelimit_test.cc:96]: @junr03 when i am running clang-tidy <file> it is giving following error. 
/home/abhay/envoy/test/extensions/filters/network/thrift_proxy/filters/ratelimit/ratelimit_test.cc:96:1: error: invalid case style for function 'TEST_F' [readability-identifier-naming,-warnings-as-errors]
TEST_F(ThriftRateLimitFilterTest, NoRoute) {
^~~~~~~
testF

**Comment by ankatare** [comment]
@junr03 Build completed with master in sync and resolved clang-tidy issues. please review.

File: test/extensions/filters/network/thrift_proxy/filters/ratelimit/ratelimit_test.cc (hunk 1 above)
Comment (ankatare) [reply test/extensions/filters/network/thrift_proxy/filters/ratelimit/ratelimit_test.cc]: done

**Pull request merged**